### PR TITLE
Convert new QEMU functions to methods with documentation

### PR DESCRIPTION
**Pull request by jakecorrenti**
Converts new functions added in #19311 to methods and adds documentation.

[NO NEW TESTS NEEDED] 

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: pkg/machine/qemu/machine.go [hunk 1]
```diff
@@ -900,11 +920,12 @@ func waitForMachine(v *MachineVM) error {
 	return nil
 }
 
-func getProxyPID(pidFilePath machine.VMFile) (int, error) {
-	if _, err := os.Stat(pidFilePath.GetPath()); errors.Is(err, fs.ErrNotExist) {
+// getProxyPID retrieves the pid from the proxy pidfile
+func (v *MachineVM) getProxyPID() (int, error) {
```
Comment (rhatdan) [inline pkg/machine/qemu/machine.go:924]: Nit: no need for get.
func (v *MachineVM) proxyPID()

File: pkg/machine/qemu/machine.go [hunk 2]
```diff
@@ -915,20 +936,22 @@ func getProxyPID(pidFilePath machine.VMFile) (int, error) {
...
+	if err := v.PidFilePath.Delete(); err != nil {
 		return err
 	}
 	return nil
 }
 
-func getVMPid(vmPidFile machine.VMFile) (int, error) {
-	vmPidString, err := vmPidFile.Read()
+// getVMPid retrieves the pid from the VM's pidfile
+func (v *MachineVM) getVMPid() (int, error) {
```
Comment (rhatdan) [inline pkg/machine/qemu/machine.go:953]: Nit: no need for get.
func (v *MachineVM) VMPID()

**Comment by rhatdan** [comment]
Sqaush your commits and get rid of "get" prefix on functions.

**Comment by rhatdan** [comment]
LGTM
/approve

**Comment by vrothberg** [comment]
> Sqaush your commits and get rid of "get" prefix on functions.

@rhatdan, I asked @jakecorrenti to break these PRs into small commits that are easier to review.  It's best practice to not throw big commits over the fence.

**Review by vrothberg — APPROVED** [review]
/lgtm
/hold

Restarted the flaked job

**Pull request merged**